### PR TITLE
review iOS SDK PubSub and fix broken links

### PR DESCRIPTION
**Pull request by drochetti**
**Notes:**

- minor formatting changes and content review
- fix API Reference broken links on iOS
- fix API Reference on Android pointing to an iOS link

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/sdk/pubsub/fragments/ios/working-api.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ iotDataManager.connectUsingALPN(withClientId: "<YOUR_CLIENT_ID>",
                        statusCallback: mqttEventCallback)
 ```
 
-You can take a look at the [API Reference](https://aws-amplify.github.io/aws-sdk-ios/docs/reference/Classes/AWSIoTDataManager.html#//api/name/connectWithClientId:cleanSession:certificateId:statusCallback:
+You can take a look at the [API Reference](https://aws-amplify.github.io/aws-sdk-ios/docs/reference/AWSIoT/Classes/AWSIoTDataManager.html#/c:objc(cs)AWSIoTDataManager(im)connectUsingALPNWithClientId:cleanSession:certificateId:statusCallback:
 ) to get more information.
```
Comment (brene) [inline docs/sdk/pubsub/fragments/ios/working-api.md:34]: superfluous "⏎"

Comment (drochetti) [reply docs/sdk/pubsub/fragments/ios/working-api.md]: my editor is configured to add a line-break add the end of the files automatically... basically this: https://stackoverflow.com/a/729795

Comment (brene) [reply docs/sdk/pubsub/fragments/ios/working-api.md]: now orries. committed. afix.

File: docs/sdk/pubsub/fragments/ios/getting-started.md [hunk 2]
```diff
@@ -17,15 +17,15 @@ When used with `AWSIoTDataManager`, PubSub is capable of signing request accordi
 The `Podfile` that you configure to install the AWS Mobile SDK must contain the `AWSIoT` pod:
 
 ```ruby
-    platform :ios, '9.0'
+platform :ios, '9.0'
```
Comment (wooj2) [inline docs/sdk/pubsub/fragments/ios/getting-started.md:20]: We should probably bump this to `11.0` ?

File: docs/sdk/pubsub/fragments/ios/getting-started.md [hunk 3]
```diff
@@ -17,15 +17,15 @@ When used with `AWSIoTDataManager`, PubSub is capable of signing request accordi
...
 
-    target :'YOUR-APP-NAME' do
-      use_frameworks!
+target :'YOUR-APP-NAME' do
+  use_frameworks!
 
-        pod  'AWSIoT', '~> 2.13.0'
-        # other pods
+  pod  'AWSIoT', '~> 2.13.0'
```
Comment (wooj2) [inline docs/sdk/pubsub/fragments/ios/getting-started.md:25]: In our IoT [example](https://github.com/awslabs/aws-sdk-ios-samples/blob/master/IoT-Sample/Swift/Podfile), we also include AWSMobileClient.  I wonder if we should also call out AWSMobileClient?

**Pull request merged**